src/views/New.vue

```diff
@@ -1,24 +1,74 @@
 <template>
-  <form class="card">
+  <form class="card" @submit.prevent="createTask">
     <h1>Создать новую задачу</h1>
     <div class="form-control">
       <label for="title">Название</label>
-      <input type="text" id="title">
+      <input type="text" id="title" v-model.trim="title">
     </div>
 
     <div class="form-control">
       <label for="date">Дата дэдлайна</label>
-      <input type="date" id="date">
+      <input type="date" id="date" v-model="date">
     </div>
 
     <div class="form-control">
       <label for="description">Описание</label>
-      <textarea id="description"></textarea>
+      <textarea id="description" v-model.trim="description"
+      ></textarea>
     </div>
 
-    <button class="btn primary">Создать</button>
+    <button class="btn primary" type="submit" :disabled="!formReady">Создать</button>
   </form>
 </template>
 
 <script>
+import {computed, ref} from 'vue'
+import {useRouter} from 'vue-router'
+
+export default {
+  setup() {
+    const url = 'https://vue3-course-work-3-default-rtdb.europe-west1.firebasedatabase.app/task.json'
+    const router = useRouter()
+    const title = ref('')
+    const date = ref(null)
+    const description = ref('')
+    let status = 'active'
+
+    const createTask = async () => {
+      const response = await fetch(url, {
+        method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify({
+          title: title.value,
+          date: new Date(date.value),
+          description: description.value,
+          status: compareDate()
+        })
+      })
+
+      const firebaseData = await response.json()
+      await router.push('/')
+    }
+
+    function compareDate() {
+      if ((new Date(date.value) - new Date) < 0) {
+        return status = 'done'
```

Comment:
И здесь тогда уже не пригодится переменная
```suggestion
        return 'done'
```